be/src/exec/pipeline/olap_chunk_source.cpp

```diff
@@ -351,35 +353,37 @@ Status OlapChunkSource::buffer_next_batch_chunks_blocking_for_workgroup(size_t b
 
     using namespace vectorized;
     int64_t time_spent = 0;
-    for (size_t i = 0; i < batch_size && !can_finish; ++i) {
-        {
-            SCOPED_RAW_TIMER(&time_spent);
-
-            ChunkUniquePtr chunk(
-                    ChunkHelper::new_chunk_pooled(_prj_iter->encoded_schema(), _runtime_state->chunk_size(), true));
-            _status = _read_chunk_from_storage(_runtime_state, chunk.get());
-            if (!_status.ok()) {
-                // end of file is normal case, need process chunk
-                if (_status.is_end_of_file()) {
-                    ++(*num_read_chunks);
-                    _chunk_buffer.put(std::move(chunk));
+    TRY_CATCH_BAD_ALLOC({
```

Comment:
TRY_CATCH_BAD_ALLOC encloses too many lines, reference to np-engine part, it just enclose only one stmt in almost     all scenarios, in rare cases, it encloses a few lines.
be/src/exec/vectorized/olap_scan_node.cpp
```
147:        TRY_CATCH_BAD_ALLOC(_fill_chunk_pool(1, first_call));
538:    TRY_CATCH_BAD_ALLOC(_fill_chunk_pool(chunks, true));
```

Reply:
chunk_pool isn't used in pipeline scan, so I remove this CHECK.